### ISSUE TITLE
xgboost working in python but not in ipynb?

### ISSUE THREAD

**Issue by Chelimin** [dmlc/xgboost#2159]
For bugs or installation issues, please provide the following information.
The more information you provide, the more easily we will be able to offer
help and advice.

## Environment info
Operating System:OSX

Package used (python/R/jvm/C++):Python

`xgboost` version used:0.6

Hi, I installed xgboost using Miniconda. It works in the python environment but does not work in Jupyter notebook. Have set the python package path. Could someone please help? Thanks! 

In python:

```
>>> import xgboost as xgb
>>> xgb
<module 'xgboost' from '/Users/chelimin/miniconda2/lib/python2.7/site-packages/xgboost.pyc'>

```
In ipynb:

```
import xgboost as xgb
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-745aa3a2d734> in <module>()
----> 1 import xgboost as xgb

ImportError: No module named xgboost
```

**Comment by gabsens** [comment]
How did you fix the issue ?

**Comment by dgkadesewa** [comment]
Same. how did you fix this?

**Comment by pancha0** [comment]
How did you fix it?

**Comment by superbobry** [comment]
I suspect the cause is that IPython uses a different Python version/installation than XGBoost.

**Comment by pancha0** [comment]
It seems that my issue is caused by using a different python kernel for jupyter notebook. 

I created a virtual environment for python 3.5. Even though it is activated, I also need to  install a kernel for this virtual environment in the notebook. See this doc for how to install python kernel for different environment:

http://ipython.readthedocs.io/en/stable/install/kernel_install.html#kernels-for-different-environments

**Comment by Chelimin** [comment]
Hi everyone, 

Sorry for the delayed response. You can fix this by pip installing in your ipython notebook:

`!pip install xgboost`

or this stackoverflow entry might help: 
https://stackoverflow.com/questions/44856105/jupyter-notebook-xgboost-import

**Comment by FaJoBenWhales** [comment]
Thanks Chelimin !! calling 
!pip install xgboost
from within Jupyter Notebook (Ubuntu, Python 3,5) solved the problem

**Comment by FaJoBenWhales** [comment]
BTW: it might be irritating, that the process triggered by 
!pip install xgboost
takes several minutes (on an I7 Notebook) without any user notification (Notebook seems frozen...).

**Comment by puifais** [comment]
This happened to me then I realized why:  I created a conda environment and installed xgboost. This worked in python shell but not in jupyter notebook because the jupyter notebook I was using was from the ananconda's folder as oppose to this anaconda environment's folder. Whoops silly me. So this is what I did to fix:

1. create conda environment
2. activate it and install jupyter notebook and xgboost
3. open jupyter notebook and use xgboost
